### ISSUE TITLE
hub pr checkout now fails if the local branch already exists

### ISSUE THREAD

**Issue by josephtannenbaum** [github/hub#2366]
**Command attempted:**

`hub pr checkout 123` 

where 123 is the number of an already-existing PR; I also already had the PR's branch on my local machine.

**What happened:**

I expected hub to just checkout my local branch, instead it failed:

`fatal: A branch named 'jt-foo-bar' already exists.`

I noticed the manpage says "Check out the head of a pull request in a new branch" which fits the behavior today, but in the past the command has acted "smartly" in this case. The [releases page](https://github.com/github/hub/releases) mentions some changes to `hub pr checkout` but not this one specifically. 

**More info:**

`hub version` output:

```
git version 2.24.0
hub version 2.13.0
```

Thank you. I really like `hub` and if this change was intended, it's disappointing but I can survive without it!

**Comment by mislav** [comment]
Hi thank you for reporting! I did not intend to change this behavior in 2.13, so checking out a PR that you already previously checked out should definitely work, and if it doesn't then that's a bug.

I cannot reproduce this, however. `hub pr checkout 123` works for me even if the target branch had already existed. The only exception is when the target branch existed, but fetching from PR's head is not a fast forward; hub will refuse to override the target branch to avoid losing your unpushed work:

```
$ hub pr checkout 123
From https://github.com/github/hub
 ! [rejected]        refs/pull/123/head -> my-head-branch  (non-fast-forward)
```

**Comment by josephtannenbaum** [comment]
Thank you for checking it out! After I opened this, I found that the error only happens sometimes, and I couldn't find any differentiator... plus recently it hasn't happened at all. May not be related to `hub`

**Comment by Rkbhasan1** [comment]
> **Command attempted:**
> 
> `hub pr checkout 123`
> 
> where 123 is the number of an already-existing PR; I also already had the PR's branch on my local machine.
> 
> **What happened:**
> 
> I expected hub to just checkout my local branch, instead it failed:
> 
> `fatal: A branch named 'jt-foo-bar' already exists.`
> 
> I noticed the manpage says "Check out the head of a pull request in a new branch" which fits the behavior today, but in the past the command has acted "smartly" in this case. The [releases page](https://github.com/github/hub/releases) mentions some changes to `hub pr checkout` but not this one specifically.
> 
> **More info:**
> 
> `hub version` output:
> 
> ```
> git version 2.24.0
> hub version 2.13.0
> ```
> 
> Thank you. I really like `hub` and if this change was intended, it's disappointing but I can survive without it!

.

**Comment by iamkirkbater** [comment]
Hi, I know I'm a little late to this party but I'm trying to figure out how to get past this?

A user submitted a PR to a repo, I pulled the PR and tested it.  There were some changes that needed to be made, and a rebase happened.

I'm attempting to pull the changes again and I'm getting the error listed here (copied from above):
```! [rejected]        refs/pull/123/head -> my-head-branch  (non-fast-forward)```

I can't actually find out _how_ to pull the newest reference here.  Everything my google-fu is turning up (which is how I got here) is telling me about _pushing_ a reference, or telling me to `git fetch` the branch, but I'm not sure what the branch would be in order to run that command.
